lib/internal/modules/cjs/loader.js

```diff
@@ -1003,26 +1004,25 @@ Module._extensions['.node'] = function(module, filename) {
   return process.dlopen(module, path.toNamespacedPath(filename));
 };
 
-Module._extensions['.mjs'] = function(module, filename) {
-  throw new ERR_REQUIRE_ESM(filename);
-};
-
 // Bootstrap main module.
-Module.runMain = function() {
+Module.runMain = function(main) {
+  // Note extension resolution for the main entry point can be deprecated in a
+  // future major.
+  const mainPath = Module._findPath(path.resolve(main), null, true);
   // Load the main module--the command line argument.
-  if (experimentalModules) {
-    asyncESM.loaderPromise.then((loader) => {
-      return loader.import(pathToFileURL(process.argv[1]).href);
-    })
-    .catch((e) => {
-      internalBinding('errors').triggerUncaughtException(
-        e,
-        true /* fromPromise */
-      );
-    });
-    return;
-  }
-  Module._load(process.argv[1], null, true);
+  asyncESM.loaderPromise.then((loader) => {
+    return loader.import(pathToFileURL(mainPath || process.argv[1]).href);
```

Comment:
IIUC, with this change there are no uncaught exceptions in CJS entry points anymore - they are all turned into handled rejections here. The source line seems to get lost once the error gets passed down through rejections this way.

Reply:
Resolved in the latest commit.